seqr/views/utils/orm_to_json_utils.py

```diff
@@ -799,65 +800,105 @@ def _get_collaborator_json(collaborator, include_permissions, can_edit, is_anvil
         })
     return collaborator_json
 
+def _get_gene_model(gene, field):
+    # prefetching only works with all()
+    return next((model for model in getattr(gene, '{}_set'.format(field)).all()), None)
 
-def get_json_for_genes(genes, user=None, add_dbnsfp=False, add_omim=False, add_constraints=False, add_notes=False,
-                       add_primate_ai=False, add_mgi=False):
+def _add_gene_model(field, return_key):
+    def _add_gene_model_func(gene):
+        model = _get_gene_model(gene, field)
+        return {return_key: _get_json_for_model(model) if model else None}
+    return _add_gene_model_func
+
+def _add_dbnsfp(gene):
+    model = _get_gene_model(gene, 'dbnsfpgene')
+    if model:
+        return _get_json_for_model(model)
+    else:
+        return _get_empty_json_for_model(dbNSFPGene)
+
+def _add_omim(gene):
+    omim_phenotypes = _get_json_for_models(gene.omim_set.all())
+    return {
+        'omimPhenotypes': [phenotype for phenotype in omim_phenotypes if phenotype['phenotypeMimNumber']],
+        'mimNumber': omim_phenotypes[0]['mimNumber'] if omim_phenotypes else None,
+    }
+
+def _add_mgi(gene):
+    model = _get_gene_model(gene, 'mgi')
+    return {'mgiMarkerId': model.marker_id if model else None}
+
+OMIM = 'omim'
+CONSTRAINT = 'constraint'
+CN_SENSITIVITY = 'cn_sensitivity'
+DBNSFP = 'dbnsfp'
+PRIMATE_AI = 'primate_ai'
+MGI_FIELD = 'mgi'
+NOTES= 'notes'
+VARIANT_GENE_DISPLAY_FIELDS = {
+    OMIM: (Omim, _add_omim),
+    CONSTRAINT: (GeneConstraint, None),
+    CN_SENSITIVITY: (GeneCopyNumberSensitivity, _add_gene_model('genecopynumbersensitivity', 'cnSensitivity')),
+}
+VARIANT_GENE_FIELDS = {
+    DBNSFP: (dbNSFPGene, _add_dbnsfp),
+    PRIMATE_AI: (PrimateAI, _add_gene_model('primateai', 'primateAi')),
+}
+VARIANT_GENE_FIELDS.update(VARIANT_GENE_DISPLAY_FIELDS)
+ALL_GENE_FIELDS = {
+    MGI_FIELD: (MGI, _add_mgi),
+    NOTES: (None, None),
+}
+ALL_GENE_FIELDS.update(VARIANT_GENE_FIELDS)
+
+def get_json_for_genes(genes, user=None, add_all=False, add_variant_gene_fields=False, add_variant_gene_display_fields=False):
```

Comment:
The `add_*` arguments are one-hot. Can it be changed to one argument for the encoding of the field types?

Reply:
I do not know what that means

Reply:
I mean that `add_all`, `add_variant_gene_fields`, and `add_variant_gene_display_fields` cannot be true at the same time.  They can be replaced by one parameter like `add_field_type` which can have a value of any one of `add_all`, `add_variant_gene_fields`, and `add_variant_gene_display_fields`. It will be less confusing.